src/d3a/d3a_core/redis_connections/redis_communication.py

```diff
@@ -66,15 +68,20 @@ def _subscribe_to_channels(self):
         self.pubsub.subscribe(**self._sub_callback_dict)
         self.pubsub.run_in_thread(sleep_time=0.5, daemon=True)
 
-    def _generate_redis_response(self, response, simulation_id, is_successful, command_type):
-        stop_response_channel = f'{simulation_id}/response/stop'
+    def _generate_redis_response(self, response, simulation_id, is_successful, command_type,
+                                 response_params=None):
+        if response_params is None:
+            response_params = {}
```

Comment:
How about assigning this default as part of method argument?

Reply:
Reason I did that is in order to circumvent the default value binding in Python methods, by using a sentinel value (None in this case). You can find a more detailed explanation here, http://effbot.org/zone/default-values.htm I highly recommend you to read this.